File: LC/Medium/128.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nset = set(nums)
        ans = 0
        
        for num in nums:
            if num - 1 not in nset:
                curr_ans = 1
                
                while num + curr_ans in nset:
                    curr_ans += 1
                
                ans = max(ans, curr_ans)  
                
        return ans

File: LC/C++/Medium/128.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int ans = 0;
        std::unordered_set<int> seen(nums.begin(), nums.end());

        for (auto num : seen) {
            if (!seen.contains(num - 1)) {
                int curr = 1;
                while (seen.contains(num + curr)) {
                    curr++;
                }
                ans = max(ans, curr);
            }
        }

        return ans;
    }
};